### PR TITLE
Issue #13345: enable example test for ParameterNumberCheck

### PR DESCRIPTION
**Pull request by Kexin2000**
Part of Issue #13345 
example1-2 diff：[https://www.diffchecker.com/8cX2lurj/](url)
example1-3 diff：[https://www.diffchecker.com/VR2DaU82/](url)
example1-4 diff：[https://www.diffchecker.com/t4ZnX1ph/](url)

### CODE REVIEW

**Review by MANISH-K-07 — COMMENTED** [review]
The issue description has been updated with few new rules.
We are requesting that java code be same in all examples and differ only by violation comments. If something is on line X in one example, it must be on the same line X in all other examples also.
Please refer ideal updates at #13345 description for reference.

**Comment by Kexin2000** [comment]
Is there something I'm missing? The 4 examples are completely the same template, only the comments and class names are different. Details can be found at the following link.
example1-2 diff：[https://www.diffchecker.com/8cX2lurj/](https://github.com/checkstyle/checkstyle/pull/url)
example1-3 diff：[https://www.diffchecker.com/VR2DaU82/](https://github.com/checkstyle/checkstyle/pull/url)
example1-4 diff：[https://www.diffchecker.com/t4ZnX1ph/](https://github.com/checkstyle/checkstyle/pull/url)

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/Example3.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  @JsonCreator
+  // violation below, 'More than 7 parameters (found 8)'
+  Example3(int a, int b, int c, int d,
+           int e, int f, int g, int h) {}
+
+  // violation below, 'More than 7 parameters (found 8)'
+  Example3(String a, String b, String c, String d,
+           String e, String f, String g, String h) {}
+
+  @interface JsonCreator {}
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/Example3.java:25]: Please move it below `// xdoc section -- end`, user should not pay attention to this declaration.

**Review by romani — CHANGES_REQUESTED** [review]
Item to improve:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/Example3.java (hunk 1 above)
Comment (Kexin2000) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/parameternumber/Example3.java]: done. any other changes needed ?
I'm not sure why the links are not working either. I had my friend try it and she were able to open them.

**Comment by MANISH-K-07** [comment]
Github, generate site

**Review by romani — APPROVED** [review]
Ok to merge.

Good job.

**Pull request merged**